apps/barista-design-system/src/shared/services/scroll-spy.service.ts

```diff
@@ -14,212 +14,123 @@
  * limitations under the License.
  */
 
-import { DOCUMENT } from '@angular/common';
-import { Inject, Injectable } from '@angular/core';
-import { Observable, ReplaySubject, Subject, fromEvent } from 'rxjs';
-import { auditTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+import { Injectable, NgZone, Inject } from '@angular/core';
+import { BehaviorSubject, Observable, fromEvent } from 'rxjs';
+import { distinctUntilChanged, auditTime } from 'rxjs/operators';
 import { Platform } from '@angular/cdk/platform';
+import { compareValues } from '@dynatrace/barista-components/core';
+import { DOCUMENT } from '@angular/common';
 
-/**
- * CODE COPIED FROM: 'https://github.com/angular/angular/blob/master/aio/src/app/shared/scroll-spy.service.ts' and modified
- */
-
-export interface BaScrollItem {
+/** Contains boundingclientrect top property and element. Used to check which item is active */
+interface HeadlineElement {
+  top: number;
   element: Element;
-  index: number;
-}
-
-export interface BaScrollSpyInfo {
-  active: Observable<BaScrollItem | null>;
-  unspy(): void;
 }
 
-/*
- * Represents a "scroll-spied" element. Contains info and methods for determining whether this
- * element is the active one (i.e. whether it has been scrolled passed), based on the window's
- * scroll position.
+/**
+ * The Scroll spy service should get the elements from outside, then check wich element is
+ * currently active and should be highlighted.
+ * Save that item into a stream and the toc component then subscribes to it.
+ * The TOC component then handles the check which item to highlight
  */
-export class BaScrollSpiedElement implements BaScrollItem {
-  top = 0;
+@Injectable({ providedIn: 'root' })
+export class BaScrollSpyService {
+  /** Top Offset */
+  OFFSET = 66;
```

Comment:
Can be moved to a constant outside of the class.
```
/**
* The offset from the top of the viewport that triggers an toc item to be active
* This enables us to have an item becoming active although its not on the very top of the viewport
* e.g. in case of a sticky navigation 0 offset would not work
*/
const ACTIVE_ITEM_TOP_OFFSET = 66;
```